x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchMaintenanceService.java

```diff
@@ -62,31 +68,29 @@ public void clusterChanged(ClusterChangedEvent event) {
         tryStartCleanup(state);
     }
 
-    void tryStartCleanup(ClusterState state) {
+    synchronized void tryStartCleanup(ClusterState state) {
```

Comment:
I think that isCleanupRunning no longer needs to be atomic boolean given that it's only accessed from synchronized methods

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/55435/commits/47f35443ad45f0fb2ac4959761c99ff195be22ac